cli/src/action/circuit/signing.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::cell::RefCell;
+use std::fs::File;
+use std::io::Read;
+
+use sawtooth_sdk::signing::{secp256k1, Context};
+
+use crate::error::CliError;
+
+thread_local! {
+    pub static CONTEXT: RefCell<secp256k1::Secp256k1Context> = RefCell::new(secp256k1::Secp256k1Context::new());
+}
+
+/// Reads a private key from the given file name.
+pub fn read_private_key(file_name: &str) -> Result<String, CliError> {
+    let mut file = File::open(file_name).map_err(|err| {
+        CliError::EnvironmentError(format!("Unable to open {}: {}", file_name, err))
+    })?;
+
+    let mut buf = String::new();
+
+    file.read_to_string(&mut buf).map_err(|err| {
+        CliError::EnvironmentError(format!("Unable to read {}: {}", file_name, err))
+    })?;
+
+    Ok(buf)
+}
+
+/// Returns the public key bytes for the given private key (hex).
+pub fn public_key_for(private_key: &str) -> Result<Vec<u8>, CliError> {
+    let key = secp256k1::Secp256k1PrivateKey::from_hex(private_key).map_err(|err| {
+        CliError::ActionError(format!("An invalid private key was provided: {}", err))
+    })?;
+
+    CONTEXT
+        .with(|ctx| ctx.borrow().get_public_key(&key))
+        .map(|pub_key| pub_key.as_slice().to_vec())
+        .map_err(|err| {
+            CliError::ActionError(format!(
+                "Unable get public key from provided private key: {}",
+                err
+            ))
+        })
+}
+
+pub fn sign_bytes(private_key: &str, bytes: &[u8]) -> Result<Vec<u8>, CliError> {
```

Comment:
Should create a Signer and pass it from around main() instead of a private_key. Remove all singing wrapper code. The only signing code should be what we need to read the key and get a signer. No thread-global variable.

Reply:
Dropped the signing commit and code.  Replaced with the use of a combination of sawtooth-sdk and splinter::signing modules